### DISCUSSION TITLE
Measure RT during several display changes.

### DISCUSSION THREAD

**Discussion by dschreij** [jspsych/jsPsych#584] [Q&A]
I have a small question, but don't know if the Github issue board is the place to ask :). My question is related to #238, but that answer seems a bit outdated (there no longer is a single-stim plugin as far as I can see).

I want to show three displays in (short) succession, and _start measuring RT from the onset of the first display_. The trial procedure should terminate at any point that the participants press one of the response keys. So if the response is given while the second image is shown, the third display should be skipped, and preferably the display of the second image is stopped as well. 
Is this possible in jsPsych? I tried the following, but no response is registered in this setup:

```javascript
var slide1 = {
  type: 'image-keyboard-response',
  stimulus: jsPsych.timelineVariable('image1'),
  trial_duration: 500
}

var slide2 = {
  type: 'image-keyboard-response',
  stimulus: jsPsych.timelineVariable('image2'),
  trial_duration: 500
}

var slide3 = {
  type: 'html-keyboard-response',
  stimulus: '<div style="font-size:60px;">AA<br>AA</div>',
  trial_duration: 2000
}

var stimulus = {
  type: 'image-keyboard-response',
  timeline: [slide1, slide2, slide3],
  choices: ['f', 'j'],
  data: jsPsych.timelineVariable('data'),
  on_finish: function(data){
    data.correct = data.key_press == jsPsych.pluginAPI.convertKeyCharacterToKeyCode(data.correct_response)
    console.log("Response was", data.correct)
  }
}
```
Could anyone point me in the right direction on how to tackle this, or tell me if it's possible at all?

**Comment by jodeleeuw** [comment]
I think you'll need to use a slightly more complicated structure where you nest each of the three slide objects on their own timeline so that you can use a `conditional_function` to check if there was a response to the previous trial before continuing to display the current trial. 

The simplest way to do the continuous RT measure is probably just to add 500 if the RT is in the second trial, and add 1000 if it's in the third. This should be pretty accurate. If you are worried about the small amount of noise that might introduce then you might get more accurate RTs by looking at the `time_elapsed` variable that is automatically added to each data entry and you could calculate how long slide1 / slide2 displayed. It should be very close to 500.

**Comment by dschreij** [comment]
Thanks for your suggestion. Maybe I will write a SVP plugin if I find the time, to make something like this easier in the future. You can close this issue if you want.
